cpp/src/arrow/compute/kernels/cast.cc

```diff
@@ -17,6 +17,9 @@
 
 #include "arrow/compute/kernels/cast.h"
 
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
+#include <boost/numeric/conversion/cast.hpp>
```

Comment:
Is it possible to not rely on Boost for this, e.g. are there some alternatives in the STL or that we can access otherwise? I will review the rest in more detail later

Reply:
Seems that `boost::numeric_cast` and `boost::lexical_cast` are not replaceable by STL.
STL has `std::to_string`, but it does not support small size ints.
http://en.cppreference.com/w/cpp/string/basic_string/to_string

Reply:
Wouldn't it be ok in the of small size ints just to upcast them? This should not affect performance as it's a small temporary.